src/mds/CDir.cc

```diff
@@ -2421,6 +2421,13 @@ void CDir::_omap_commit_ops(int r, int op_prio, int64_t metapool, version_t vers
   auto commit_one = [&](bool header=false) {
     ObjectOperation op;
 
+    /*
+     * just skip submiting the invalid request, which could cause
+     * the cephfs to become readonly.
+     */
+    if (!header && _set.empty() && _rm.empty())
+      return;
```

Comment:
This should be unreachable with your changes, no? I guess it's better than an assert...

Reply:
Yeah, with the change below it shouldn't. Make sense and I will revise it.